e2e-tests/e2e.bats

```diff
@@ -96,11 +96,23 @@ kwctl() {
     [ $(expr "$output" : '.*Policy successfully verified.*') -ne 0 ]
     kwctl verify -k test-data/sigstore/cosign3.pub registry://ghcr.io/kubewarden/tests/pod-privileged:v0.1.9
     [ "$status" -eq 1 ]
-    [ $(expr "$output" : '.*No signing keys matched given constraints*') -ne 0 ]
+    [ $(expr "$output" : '.*Image verification failed: allOf list: No Signature Layer passed verification.*') -ne 0 ]
+    # TODO this shouldn't mention allOf, user has no notion of it
```

Comment:
In this case, it's doing `kwctl verify -k test-data/sigstore/cosign3.pub`. And then we internally create a VerificationSettings that we pass to policy-fetcher.
I think the user would be better served getting a list on which constraints are not satisfied, instead of the `allOf` message.

Reply:
Solved by https://github.com/sigstore/sigstore-rs/pull/42.